provider/pkg/provider/deployment_settings.go

```diff
@@ -14,14 +14,16 @@ import (
 	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 )
 
+// This is a value for imported secrets, to hint that value needs to be replaced
+// in generated code
+const replaceMe = "<REPLACE WITH ACTUAL SECRET VALUE>"
+
 type PulumiServiceDeploymentSettingsInput struct {
 	pulumiapi.DeploymentSettings
 	Stack pulumiapi.StackName
 }
 
-const FixMe = "<value is secret and must be replaced>"
-
-func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap() resource.PropertyMap {
+func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap(plaintextSettings *pulumiapi.DeploymentSettings, oldCipherSettings *pulumiapi.DeploymentSettings, isInput bool) resource.PropertyMap {
```

Comment:
Can we rename these variables just so the logical choices we're making later on are clearer? From what I understand, `plaintextSettings` is actually `oldInputs`? And `oldCipherSettings` is `oldOuputs`?

```suggestion
func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap(plaintextSettings *pulumiapi.DeploymentSettings, oldCipherSettings *pulumiapi.DeploymentSettings, isInput bool) resource.PropertyMap {
```

Reply:
Hmm, I think we should call objects of type DeploymentSettings `somethingSettings`. I don't think the method needs to know where these come from, just that one has plaintext in them and other has previous ciphertext. Maybe I can rename `old` in `oldCipherSettings` into `previous` or `last`, would that help? Or even `currentCiphertext`, to align closer with the [design](https://docs.google.com/document/d/1msoP-xNPGKOxGzTfvtYSANY_6T0NtiJHNpj6l5A2Y9g/edit#heading=h.qocf1eqtb2j9)

Reply:
I can also add booleans up top to avoid repeating check of `plaintextSettings == nil`, I think that would make it more clear

Reply:
Yeah this function was already pretty unwieldy to start with and I find it impossible to reason through some of the decisions we're making now. To me, it doesn't matter what has plaintext or what has ciphertext. The decisions are based on whether or not previous inputs or outputs are present or not. 

Honestly not sure what would make it better, I can understand it's subjective what makes more sense to me vs what makes more sense to you. At the very least can you add comments to explain what the code is doing?

Reply:
Yeah, agree this function is a mess, I can maybe refactor it later and split it up.
`To me, it doesn't matter what has plaintext or what has ciphertext.` heh, well you kinda need to know that to extract the right data

Sounds good, I will add comments on how the parameters are used and what they mean